### PR TITLE
[SIG-3339] Remove container button

### PR DESCRIPTION
**Pull request by MauritsR**
### Changes

- Adds 'remove' button to container list in map selection panel

### CODE REVIEW

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    return `data:image/svg+xml;base64,${btoa(icon)}`;
+  }, [featureTypes]);
+
+  return (
+    <List className={className} data-testid={NAME}>
+      {selection.map(({ id, description, type }) => (
+        <ListItem data-testid={`${NAME}Item-${id}`} key={id} tabIndex={-1}>
+          <StyledItemSpan>
+            <StyledIcon url={getIconUrl(type)} />
+            {description} - {id}
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:80]: Please do the formatting in a format string

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  display: inline-block;
+  margin-right: ${themeSpacing(2)};
+  flex-shrink: 0;
+  width: ${DEFAULT_ICON_SIZE}px;
+  height: ${DEFAULT_ICON_SIZE}px;
+  background-image: url(${({ url }) => url});
+  background-size: cover;
+`;
+
+const StyledItemSpan = styled.span`
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:48]: I would use a div here as is a block so far as I can see.

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  align-items: center;
+  justify-content: space-between;
+  font-size: initial;
+
+  &:focus {
+    outline-style: none;
+  }
+`;
+
+const StyledIcon = styled.span<{ url: string }>`
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:38]: Please use the `Icon` component from `asc` here.

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+`;
+
+export interface ContainerEditListProps {
+  selection: Item[];
+  onRemove: (id: string) => void;
+  featureTypes: FeatureType[];
+  className?: string;
+}
+
+const ContainerEditList: FunctionComponent<ContainerEditListProps> = ({
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:61]: There is a lot of overlap with `ContainerList` in this component. Lets merge the two. When onRemove is not passed then we don't show the `Close` icon.

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+const StyledButton = styled(Button)`
+  flex-shrink: 0;
+  box-shadow: 0 0 0 2px rgba(0, 0, 0, 0.1);
+`;
+
+const List = styled.ul`
+  padding: 0;
+`;
+
+const ListItem = styled.li`
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:25]: Can we use the existing List or IconList component here?

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+import type { FunctionComponent } from 'react';
+import styled from 'styled-components';
+
+import { themeSpacing } from '@amsterdam/asc-ui';
+import { Close } from '@amsterdam/asc-assets';
+
+import Button from 'components/Button';
+import type { FeatureType, Item } from '../../types';
+
+const DEFAULT_ICON_SIZE = 40;
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:11]: This value is declared somewhere as wel.

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+const REMOVE_BUTTON_SIZE = 32;
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:12]: I think these constants are not adding to much, till now in the code the icon sizes are not specified as constants. I would keep that style for this kind of constants.

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+const REMOVE_ICON_SIZE = 12;
+const NAME = 'containerEditList';
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx:14]: Moving the data-testid to a constant does not fit the coding style used in this project, the cases when a variable is used are when they are passed as props and have some other usage besides testid. Shell we keep consistency here?

**Review by aditudorache — CHANGES_REQUESTED** [review]

File: src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx (hunk 5 above)
Comment (MauritsR) [reply src/signals/incident/components/form/ContainerSelect/components/ContainerEditList/index.tsx]: Yes - didn't know those were supplied by Amsterdam Styled Components

File: src/signals/incident/components/form/ContainerSelect/components/Selector/index.tsx [hunk 9]
```diff
@@ -178,32 +182,28 @@ const Selector = () => {
...
-                    <Paragraph as="h6">Maak een keuze op de kaart</Paragraph>
-                  )}
-                </ButtonBar>
-
-                <ButtonBar>
-                  <StyledButton onClick={close} variant="primary">
-                    Meld deze container{selection.length > 1 ? 's' : ''}
-                  </StyledButton>
-                </ButtonBar>
+                <Button onClick={addContainer}>Containers toevoegen</Button>
```
Comment (aditudorache) [inline src/signals/incident/components/form/ContainerSelect/components/Selector/index.tsx:192]: Can be removed.

**Review by aditudorache — CHANGES_REQUESTED** [review]
Nice work! I would remove the test addContainer button as well while we have implemented that functionality

File: src/signals/incident/components/form/ContainerSelect/components/Selector/index.tsx (hunk 9 above)
Comment (MauritsR) [reply src/signals/incident/components/form/ContainerSelect/components/Selector/index.tsx]: done!

**Pull request merged**